### ISSUE TITLE
Database primary key reached maximum value of sequence

### ISSUE THREAD

**Issue by zkonge** [spiffe/spire#5501]
* **Version**: 1.10.4 (and order)
* **Platform**: Linux super-dev 6.1.0-25-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.106-3 (2024-08-26) x86_64 GNU/Linux
* **Subsystem**: server

## What happened
Our psql raised the "error reached maximum value of sequence" by `node_resolver_map_entries_id_seq`. And server down.

We have a large amount of agents. and some of the agents we deploy are constantly restarting,

## Why
https://github.com/spiffe/spire/blob/3d158ce5635ab92fc5869b4ecda44c6a153b2a06/pkg/server/datastore/sqlstore/models.go#L10
It is seems that the `ID` field in `node_resolver_map_entries` is not enough in such scenarios.

SPIRE, which uses gorm v1, will mapping the `uint` to `int` type in DB and `uint64` to `bigint` (By the way, gorm v2 will mapping both `uint` and `uint64` to `bigint`).

Each agent restarting (or attestation) will increase the `node_resolver_map_entries_id_seq` value.

## What to do
Use uint64 in Model primary key
Modify the type?

I'm worried about the DDL modification in the migration, for a large table, the type modification will lock the whole table and the server will down.

**Comment by azdagron** [comment]
I don't think we have a choice other than migrating. We'll have to be careful with our use of AutoMigrate though. We may need to investigate if gorm v1 has a way to tag the ID to use a bigint instead of int (maybe declaring our own model type that defines ID as uint64 or something).

**Comment by zkonge** [comment]
**For anyone who wants to modify the database schema directly to expand the primary key ID range:**

**At the moment, this cannot be done without a source change.**

In `sqlstore.go`
https://github.com/spiffe/spire/blob/941366df9b909d9c7382911463a085a4f206b538/pkg/server/datastore/sqlstore/sqlstore.go#L1998

SPIRE currently parses pagination tokens using 32-bit unsigned integers. To support larger IDs, all occurrences of:
```
strconv.ParseUint(<PaginationToken>, 10, 32)
```
need to be changed to:
```
strconv.ParseUint(<PaginationToken>, 10, 64)
```


## Background
I’m testing SPIRE with *CockroachDB*, which uses BIGINT primary keys and generates IDs via unique_rowid() for distributed/sharded workloads. As a result, row IDs can be much larger than 32-bit. For example:

`1142618514430656513`

With the current bitSize=32 parsing, such values cannot be represented and will fail when used as pagination tokens.

Any update on this issue?
Seems the modification is not quite complex.
